src/main/java/com/example/controller/NasaPicturesController.java

```diff
@@ -0,0 +1,31 @@
+package com.example.controller;
+
+
+import com.example.exception.PictureNotFoundException;
+import com.example.service.NasaPictureService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.Optional;
+
+import static org.springframework.http.HttpHeaders.LOCATION;
+import static org.springframework.http.HttpStatus.MOVED_PERMANENTLY;
+
+@RestController
+@RequestMapping("/mars/pictures")
+@RequiredArgsConstructor
+public class NasaPicturesController {
+
+    private final NasaPictureService nasaPictureService;
+
+    @GetMapping("/{sol}/largest")
+    public ResponseEntity<?> getLargestPicture(@PathVariable Integer sol, @RequestParam Optional<String> camera) {
```

Comment:
A wildcard in the return type is not a good practice. You can use ResponseEntity<Void> here, as the current response doesn't contain a body.